tokio/src/sync/broadcast.rs

```diff
@@ -820,6 +876,11 @@ fn new_receiver<T>(shared: Arc<Shared<T>>) -> Receiver<T> {
     assert!(tail.rx_cnt != MAX_RECEIVERS, "max receivers");
 
     tail.rx_cnt = tail.rx_cnt.checked_add(1).expect("overflow");
+    if tail.closed {
+        // Potentially need to re-open the channel, if a new receiver has been added between calls
+        // to poll()
+        tail.closed = false;
+    }
```

Comment:
This causes the test `tests/sync_broadcast::resubscribe_to_closed_channel` to fail because it also reopens the channel when the sender got dropped before.
Maybe instead of checking for `tail.closed` to reopen the channel we could check for `tail.rx_cnt == 0` before the call to `tail.rx_cnt.checked_add(1)`? In that case the channel would only be reopened if it got closed due to no more receivers being "alive". 
I only ran the sync tests locally real quick and got no failures, so maybe its something to work with?

Reply:
Thanks for the pointers! I've been meaning to get back to this and study the code a bit more, but have had some stuff crop up in my personal life so haven't had time, sorry if I'm causing unnecessary work for anyone

Reply:
No worries, there isn't a rush from our side. Things take the time they take.